credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -147,6 +148,42 @@ class Credentials(
     }
 
 
+    /**
+     * Verify and return profile from a
+     * [Selective Disclosure Response JWT](https://github.com/uport-project/specs/blob/develop/messages/shareresp.md).
+     *
+     * @param token **REQUIRED** The JWT response token from a selective disclosure request
+     *
+     * @return a [UportProfile] object
+     */
+    @Suppress("TooGenericExceptionCaught")
+    suspend fun verifyDisclosure(token: String): UportProfile {
+
+        val (_, payload, _) = JWTTools().decode(token)
+
+        val valid = mutableListOf<JwtPayload>()
+        val invalid = mutableListOf<String>()
+
+        payload.verified?.forEach {
+            try {
+                valid.add(JWTTools().verify(it))
+            } catch (e: Exception) {
```

Comment:
This should catch only exceptions that indicate the JWT is invalid.
Network errors for example don't make the JWT less valid and should be thrown.

Reply:
I used this operation initially and it caused the entire operation to be killed if one token throws an uncaught error and I felt it would defeat the general aim. What do you think? @mirceanis

Reply:
Well the point is to separate concerns, and to provide accurate info. 
If there is a network error then a token is marked as invalid which will be a false statement in most situations, making this method essentially unreliable